File: back-end/src/main/java/io/kare/backend/payload/data/MachineType.java
package io.kare.backend.payload.data;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum MachineType {
    BARBELL("BARBELL"),
    DUMBBELL("DUMBBELL"),
    CABLE("CABLE"),
    MACHINE("MACHINE"),
    BODYWEIGHT("BODYWEIGHT");

    private final String value;

    MachineType(String value) {
        this.value = value;
    }

    @JsonCreator
    public static MachineType fromString(String value) {
        for (MachineType machineType : MachineType.values()) {
            if (machineType.value.equalsIgnoreCase(value)) {
                return machineType;
            }
        }
        throw new IllegalArgumentException("Invalid MachineType: " + value);
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}

File: android/app/src/main/java/com/koleff/kare_android/data/model/dto/MachineType.kt
package com.koleff.kare_android.data.model.dto

enum class MachineType(val machineId: Int) {
    DUMBBELL(1),
    BARBELL(2),
    MACHINE(3),
    NONE(-1);

    companion object {
        fun fromId(id: Int): MachineType =
            values().find { it.machineId == id } ?: NONE
    }
}